crates/compression/src/ports.rs

```diff
@@ -77,6 +77,22 @@ where
     }
 }
 
+/// Get the whole temporal registry root
+#[cfg(feature = "fault-proving")]
+pub trait GetRegistryRoot {
+    fn registry_root(&self) -> anyhow::Result<fuel_core_types::fuel_tx::Bytes32>;
```

Comment:
From my perspective of ports and adapter, on each module we define ports for interactions that we need in order to make this module works, we don't define ports for exposing this modules functionnalities, this is done when implementing the adapters through the port traits of modules that actually wants it.

Reply:
yup, this will get refactored with work after https://github.com/FuelLabs/fuel-core/pull/2819